### PR TITLE
Removing decoding from SelfHost

### PR DESCRIPTION
**Pull request by psibernetic**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/NancyFx/Nancy/pulls) open
- [x] I have verified that I am following the Nancy [code style guidelines](https://github.com/NancyFx/Nancy/blob/45238076ad0b7f6ecabd6bae8469e30458d02efe/CONTRIBUTING.md#style-guidelines)
- [ ] I have provided test coverage for my change (where applicable) 

### Description
This is a fix for #2461.

Mr. @khellang, your will be done.

- Removing UrlDecode from relativeUrl resolution in Nancy Self Host
- Removing trailing comma  that bugged the crap out of me

### CODE REVIEW

File: src/Nancy.Hosting.Self/NancyHost.cs [hunk 1]
```diff
@@ -260,8 +260,8 @@ private Request ConvertRequestToNancyRequest(HttpListenerRequest request)
                 HostName = request.Url.Host,
                 Port = request.Url.IsDefaultPort ? null : (int?)request.Url.Port,
                 BasePath = baseUri.AbsolutePath.TrimEnd('/'),
-                Path = HttpUtility.UrlDecode(relativeUrl),
-                Query = request.Url.Query,
+                Path = relativeUrl,
```
Comment (khellang) [inline src/Nancy.Hosting.Self/NancyHost.cs]: Can we just kill the `relativeUrl` variable and do `baseUri.MakeAppLocalPath(request.Url)` directly here? :smile:

Comment (psibernetic) [inline src/Nancy.Hosting.Self/NancyHost.cs]: Guess I should turn my squiggles for that back on. -_-

**Comment by jchannon** [comment]
Are we need to do all hosts as discussed here? https://github.com/NancyFx/Nancy/issues/1280

**Comment by khellang** [comment]
No, it's only the self-host that is decoding before passing the URL to Nancy. See #2117.

OWIN is also decoding, but that's part of the spec and there's nothing we can do about it :disappointed:

**Comment by thecodejunkie** [comment]
@khellang can you add this to the 2.x upgrade doc on the wiki? Think it's important that we don't forget about it

**Pull request merged**